platform-validation/scripts/aziot-compatibility.sh

```diff
@@ -501,58 +547,75 @@ check_shared_library_dependency() {
     # IOTEDGE_COMMON_SHARED_LIBRARIES is for both aziot-edged and aziot-identityd
     for lib in $SHARED_LIBRARIES; do
         # check dependencies for `ldconfig` and fall back to `find` when its not possible
-        if [ "$(id -u)" -ne 0 ] || [ "$(need_cmd ldconfig)" != 0 ]; then
-            found_library=0
-            for path in $SHARED_LIB_PATH; do
-                if [ ! -e "$path" ]; then
-                    wrap_warning "Path : $path does not exist, Searching for other paths"
-                    continue
-                else
-                    if [ ! "$(find "$path" -name "$lib" | grep .)" ]; then
-                        found_library=0
-                    else
-                        found_library=1
-                        wrap_pass "$lib"
-                        break
-                    fi
-                fi
-            done
-            if [ $found_library -eq 0 ]; then
-                wrap_fail "$lib"
-                check_shared_library_dependency_display_util "$lib"
-            fi
+        if [ "$(id -u)" -ne 0 ]; then
+            legacy_find_and_report_libs "$lib"
         else
-            ret=$(ldconfig -p | grep "$lib")
-            if [ -z "$ret" ]; then
-                check_shared_library_dependency_display_util "$lib"
+            ret=$(need_cmd ldconfig)
+            if [ "$?" -ne 0 ]; then
+                legacy_find_and_report_libs "$lib"
             else
-                wrap_pass "$lib"
+                ret=$(ldconfig -p | grep "$lib")
+                if [ -z "$ret" ]; then
+                    display_missing_library_warning "$lib"
+                else
+                    wrap_pass "$lib"
+                fi
             fi
         fi
     done
 }
 
-check_shared_library_dependency_display_util() {
+check_storage_space() {
+    storage_path=$1
+    application_size=$2
+    buffer=$3
+
+    # Check dependencies
+    if ! need_cmd df; then
+        wrap_warn "check_storage_space"
+        wrap_warning "Could not find df utility to calculate disk space, Skipping the check"
+        return
+    fi
+
+    if ! echo "$application_size" | grep -Eq '^[0-9]+.?[0-9]+$'; then
```

Comment:
You should probably escape the '.' in the regex so it's interpreted as a period, not as "any character".

Reply:
Done